### PR TITLE
Flip expiration times

### PR DESCRIPTION
**Pull request by sophiebits**
Now:

0 - NoWork
1 - Never
...
big - low pri things
even bigger - hi pri things
max 31-bit int - Sync

Per @sebmarkbage in https://github.com/facebook/react/pull/13904.

### CODE REVIEW

**Comment by sophiebits** [comment]
Want me to make them all negative? 🙄

min 31-bit int - Sync
large negative - hi pri things
less-large negative - low pri things
...
–1 - Never
0 - NoWork

Then 0 is in the right place and it goes in the right direction…

**Comment by sebmarkbage** [comment]
I think we can find a mental model where this makes sense. E.g. a count down until the app stops working (NoWork).

**Comment by acdlite** [comment]
I’m cool with this if we rename the field, too. Like timeRemaining or something with “priority” in it.

File: packages/react-reconciler/src/ReactFiberScheduler.js [hunk 1]
```diff
@@ -855,18 +840,10 @@ function resetChildExpirationTime(
...
         newChildExpirationTime = childUpdateExpirationTime;
       }
-      if (
-        newChildExpirationTime === NoWork ||
-        (childChildExpirationTime !== NoWork &&
-          childChildExpirationTime < newChildExpirationTime)
-      ) {
+      if (childChildExpirationTime > newChildExpirationTime) {
         newChildExpirationTime = childChildExpirationTime;
       }
```
Comment (sophiebits) [inline packages/react-reconciler/src/ReactFiberScheduler.js]: At this point we could even go so far as to do

```
newChildExpirationTime = Math.max(
  newChildExpirationTime,
  child.expirationTime,
  child.childExpirationTime,
);
```

so simple… wow

doge meme

Comment (sebmarkbage) [reply packages/react-reconciler/src/ReactFiberScheduler.js]: 😢

**Comment by sophiebits** [comment]
I kinda like the negative clock (that becomes less negative over time) actually… what about you two?

**Comment by sebmarkbage** [comment]
Nah. Negative is weird. That's not Rust/WASM/et al friendly since we'd want the unsigned form there.

**Comment by acdlite** [comment]
If we didn’t already have something called “priority” I would say let’s rename it to that and merge your PR as is.

I don’t like the “expirationTime” naming regardless because it only acts like an expiration time when we schedule the callback with Scheduler. Everywhere else it acts like a relative priority.

Precdence? Urgency? Rank? (I’m just reading the thesaurus)

**Comment by sophiebits** [comment]
"Threshold" maybe.

**Comment by acdlite** [comment]
Importance?

**Comment by sebmarkbage** [comment]
Note that we also plan to add another public priority API which is not related to time (the priority of fallbacks returning to their unsuspended state).

**Comment by gaearon** [comment]
Urgency?

**Comment by sophiebits** [comment]
We're hopeless.

**Comment by gaearon** [comment]
Oh I missed @acdlite already suggested urgency.

**Comment by acdlite** [comment]
@sebastian yeah and I also don’t think we understand the relationship between the two concepts yet, either.

**Comment by sophiebits** [comment]
restlessness

**Comment by NE-SmallTown** [comment]
> If we didn’t already have something called “priority” I would say let’s rename it to that and merge your PR as is.
> I don’t like the “expirationTime” naming regardless because it only acts like an expiration time when we schedule the callback with Scheduler. Everywhere else it acts like a relative priority.

Agree, we use priority naming before but then use expirationTime to abstract too many things for processing starvation and competition better. But just from a developer perspective, the expirationTime is confused and hard to understand(lower means higher, higher means lower), and when you compare it to some names like `xxxxWork`, it's weird(a time to compare a work, what's that, oh, they are just numbers...)

**Pull request merged**